### ISSUE TITLE
Runtime race conditions

### ISSUE THREAD

**Issue by PeterTh** [insieme/insieme#17]
A collection issue to gather runtime race conditions.

**Comment by philippgs** [comment]
The integration test omp/scr/lu_reduction just crashed on hudson:
```
Command: /var/lib/jenkins/jobs/insieme_base_release_gcc-4.7.3/workspace/test/omp/scr/lu_reduction/lu_reduction.insieme.run

IRT Assertion failure in /var/lib/jenkins/jobs/insieme_base_release_gcc-4.7.3/workspace/code/utils/../runtime/include/impl/work_group.impl.h#132:
Insieme Runtime Error received (thread 0x7fd478637700): IRT_ERR_INTERNAL
Additional information:
irt_wg_get_wi_membership: membership not found for wi in wg
```

**Comment by philippgs** [comment]
Jenkins **insieme_ext_release** build no 167 failed because of a segmentation fault in **base/omp/oddEvenSort** (**insiemecc_run_c_execute** step)

**Comment by peterz-dps** [comment]
My current work on the runtime event system (see #47) also fixes at least two additional race conditions. These changes will pushed as soon as I'm done with it - probably early next week. Let's see whether this will fix any of the race conditions we experienced so far (which is always hard to tell obviously)

**Comment by philippgs** [comment]
Jenkins' <b>github_release_nightly</b> build no 73 failed because of a segmentation fault in <b>base/omp/barrier</b> (<b>insiemecc_run_c_execute</b> step)

**Comment by peterz-dps** [comment]
Jenkins' insieme_base_release-with-asserts_gcc-4.8.3_no-ocl_nightly build no 19 failed because of a race condition in the asteroidea stacks.

**Comment by peterz-dps** [comment]
The race condition mentioned above ^^ when using asteoridea stacks is supposed to be fixed with commit 1c8a3809cbf0a880ea2aa729d6fa541aa16f692b

**Comment by peterz-dps** [comment]
The Jenkins job "github_release_nightly" no. 92 segfaulted in omp/barrier. I could already reproduce a segfault for this test on my local machine. Currently investigaing...

**Comment by peterz-dps** [comment]
The Jenkins job "insieme_ext_release" no. 331 failed with an unsuccessful verification in omp/nas/cg...

**Comment by peterz-dps** [comment]
The race condition which manifested itself in omp/barrier has been fixed by e9d4052461edd62fd6bae9025e2a6f8a133c26f7

We decided to close this issue with every fixed race condition, and re-open it again once another one surfaces.
